### PR TITLE
VBLOCKS-1035 Fixing build

### PR DESCRIPTION
**Pull request by charliesantos**
**Contributing to Twilio**

> All third party contributors acknowledge that any contributions they provide will be made under the same open source license that the open source project is provided under.

- [x] I acknowledge that all my contributions will be made under the project's license.

## Pull Request Details

Our build process is failing on node 16 due to outdated dependencies. The 2 main big ones are typescript and typedoc. However, updating typedoc to the working version for our project requires changing how we document most of our files, including a breaking change where we must change how we export namespaces and classes. The latest versions of typedoc does not handle declaration merging very well. Details below.
- Update typedoc, update docstrings, do not update exports - works code wise, but breaks documentation. See attached screenshot below.
- Update typedoc, update docstrings, update exports - documentation is fixed, but we now have breaking changes due to to difference in exports

You can see my progress here https://github.com/twilio/twilio-voice.js/compare/VBLOCKS-1035-update-deps

For this PR, I'm fixing the minimum required to get the build and pipelines working, which includes:
- Adding @types/mime 3.0.0. Later versions is breaking the build. The breakage happens because we don't have a lock file which causes peerDependencies to pull in the latest versions.
- Adding lock file to prevent issues mentioned above.
- Disabling docker tests. They are skipped on firefox and broken on chrome. We will address this on a separate ticket.
- Fixing npm/build badges

For future work, we need to bump the major version in order to update all dependencies to latest version.

EDIT: I forgot to mention, the build was failing when removing `legacy-peer-deps` flag. Adding it back the build only fails due to the @types/mime issue I mentioned above. So we still need to keep `legacy-peer-deps`, unfortunately.

Screen Shot 2022-09-06 at 1 48 22 PM

### CODE REVIEW

**Comment by mhuynh5757** [comment]
By major version, do you mean `x` in `x.y.z`?

**Comment by charliesantos** [comment]
> By major version, do you mean `x` in `x.y.z`?

Yes. Basically VoiceJS 3.x 😢

File: .circleci/images/docker-compose.yml [hunk 1]
```diff
@@ -30,7 +30,7 @@ services:
       - /opt/app/node_modules
   integrationTests: # runs integration tets. Expects that sources are mounted.
     <<: *runtimeDefaults
-    command: bash -c "npm install --no-optional --no-legacy-peer-deps && npm run build && ls -la /root && ls -la /root/.npm && npm run test:network"
+    command: bash -c "npm install --legacy-peer-deps && npm run build && ls -la /root && ls -la /root/.npm && npm run test:network"
```
Comment (mhuynh5757) [inline .circleci/images/docker-compose.yml:33]: This whole file probably needs reworking.

File: .circleci/images/docker-compose.yml [hunk 2]
```diff
@@ -30,7 +30,7 @@ services:
       - /opt/app/node_modules
   integrationTests: # runs integration tets. Expects that sources are mounted.
```
Comment (mhuynh5757) [inline .circleci/images/docker-compose.yml:31]:
```suggestion
  integrationTests: # runs integration tests. Expects that sources are mounted.
```

File: .gitignore [hunk 3]
```diff
@@ -2,7 +2,6 @@
 .idea
 .nyc_output/
 config.yaml
-package-lock.json
```
Comment (mhuynh5757) [inline .gitignore:5]: 🎉

**Review by mhuynh5757 — APPROVED** [review]

**Pull request merged**